recaf-ui/src/main/java/me/coley/recaf/ui/pane/pe/PEExplorerPane.java

```diff
@@ -91,28 +95,34 @@ public void onUpdate(FileInfo newValue) {
 		this.fileInfo = newValue;
 		CadesBufferStream stream = new CadesBufferStream(newValue.getValue());
 		LittleEndianReader reader = new LittleEndianReader(stream);
-		pe = ImagePeHeaders.read(reader);
+		pe = PeImage.read(reader).getOk();
 
 		// Remove the export directory option if the file doesn't have one
-		if (dummyRoot.getChildren().contains(itemExportDirectory) && pe.getCachedExports() == null) {
+		if (dummyRoot.getChildren().contains(itemExportDirectory) && pe.exports == null) {
 			dummyRoot.getChildren().remove(itemExportDirectory);
 		}
 
 			// Reset section headers
-		ImageSectionHeader[] sectionTable = pe.sectionHeaders;
+		ArrayList<ParseResult<ImageSectionHeader>> sectionHeaders = pe.sectionHeaders;
 		itemSectionHeaders.getChildren().clear();
-		for (ImageSectionHeader header : sectionTable) {
-			TreeItem<String> sectionItem = new TreeItem<>(header.getName());
-			itemSectionHeaders.getChildren().add(sectionItem);
+		for (ParseResult<ImageSectionHeader> headerResult : sectionHeaders) {
+			if (headerResult.isOk()) {
+				TreeItem<String> sectionItem = new TreeItem<>(headerResult.getOk().getName());
+				itemSectionHeaders.getChildren().add(sectionItem);
+			}
 		}
 
 		// Add libraries to import directory
-		for (int i = 0; i < pe.getNumCachedImports(); i++) {
-			CachedLibraryImports cachedLibraryImport = pe.getCachedLibraryImport(i);
-			TreeItem<String> libraryItem = new TreeItem<>(cachedLibraryImport.getName());
-			itemImportDirectory.getChildren().add(libraryItem);
+		if (pe.exports.isOk()) {
+			LibraryExport libraryExport = pe.exports.getOk();
+			for (int i = 0; i < libraryExport.entries.length; i++) {
+				ExportEntry exportEntry = libraryExport.entries[i];
+				TreeItem<String> libraryItem = new TreeItem<>(exportEntry.name);
+				itemImportDirectory.getChildren().add(libraryItem);
+			}
```

Comment:
The diff in this section seems to change behavior. I've not dug into the new version of JavaPeParser to find the equivalent new behavior, but at a glance this seems to swap imports and exports.

Reply:
Yeah, the new line no. 122 is adding `exportEntry.name` to the control `itemImportDirectory`. Seems like a simple hiccup in the refactoring.

Aside from slight renaming, The only changed behavior in my new API is that many values are replaced with a `Result` class, and you have to call `getOk()` or `getOkOrDefault(null)`.
*(You can also print/display `peImage.warnings` to see comments from the parser)*

Reply:
Fixed